TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -42,6 +42,94 @@
 
 using namespace utest::v1;
 
+#if defined(__NEWLIB_NANO)
+
+#define MAX_STRING_SIZE 100
+//Global buffer used by test cases to store string literal
+char expected_string[MAX_STRING_SIZE];
+
+typedef enum {
+    RADIX_10,//radix base for decimal
+    RADIX_16 //radix base for Hexadecimal
+} radix_base;
+
+// This function is similar to itoa but it can convert any data type with decimal and hexadecimal numerical value into a string
+static void convert_to_string(unsigned long long  value,  char str[],  radix_base radix = RADIX_10, bool sign = false)
```

Comment:
```suggestion
static char* convert_to_string(
    uint64_t value,
    char *str,
    BaseType base = BASE_10,
    bool signed = false
)
```
Why not returning the string like `itoa()` does?